software-mansion/react-native-reanimated
# feat: Use new logger and ReanimatedError for lib warnings and errors

## Summary

This PR replaces all `console.warn`, `console.error` and `throw new Error` statements with custom logger calls (`logger.warn`, `logger.error` and `logger.newError` respectively).

It declares the logger object as a global variable for easier usage and to prevent circular imports (as it imports `runOnJS`, which was the cause of circular dependencies).

## Test plan

- open the `EmptyExample` in the example app,
- change the `SHOW_EXAMPLE` constant to see the respective error/warning,
- see better error logs with code frame that was the culprit (if possible - sometimes we cannot get pretty error stack trace, especially when functions are called asynchronously)
